tests/test_qsim_device.py

```diff
@@ -366,6 +379,37 @@ def test_expval_single_wire_with_parameters(
 
         assert np.isclose(res, expected_output, **tol)
 
+    @pytest.mark.parametrize(
+        "observable1,observable2",
+        [
+            (qml.Identity, qml.PauliZ),
```

Comment:
It would be nice to test a single `qml.Identity` too, as that is a separate case in the condition used in the `expval` method.